libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioClientWrapper.cs

```diff
@@ -13,15 +14,33 @@ namespace Microsoft.Bot.Builder.Adapters.Twilio
     public class TwilioClientWrapper
     {
         /// <summary>
-        /// Initializes the Twilio client with a user name and password.
+        /// Initializes a new instance of the <see cref="TwilioClientWrapper"/> class.
         /// </summary>
-        /// <param name="username">The user name for the Twilio API.</param>
-        /// <param name="password">The password for the Twilio API.</param>
-        public virtual void LogIn(string username, string password)
+        /// <param name="options">An object containing API credentials, a webhook verification token and other options.</param>
+        public TwilioClientWrapper(TwilioAdapterOptions options)
         {
-            TwilioClient.Init(username, password);
+            Options = options ?? throw new ArgumentNullException(nameof(options));
+
+            if (string.IsNullOrWhiteSpace(options.TwilioNumber))
+            {
+                throw new ArgumentException("TwilioNumber is a required part of the configuration.", nameof(options));
```

Comment:
Here, the exception should name the exact parameter as we did on Webex. The same for the rest of the exceptions in this method.
```suggestion
                throw new ArgumentException(nameof(options.TwilioNumber));
```